### PR TITLE
UI_process() didn't generate errors

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] CLA is signed

##### Description of change

Since there are many parts of UI_process() that can go wrong, it isn't
very helpful to only return -1 with no further explanation.  With this
change, the error message will at least show which part went wrong.

### CODE REVIEW

File: crypto/ui/ui_lib.c [hunk 1]
```diff
@@ -427,9 +427,13 @@ static int print_error(const char *str, size_t len, UI *ui)
 int UI_process(UI *ui)
 {
     int i, ok = 0;
+    const char *state = NULL;
 
-    if (ui->meth->ui_open_session && !ui->meth->ui_open_session(ui))
-        return -1;
+    if (ui->meth->ui_open_session && !ui->meth->ui_open_session(ui)) {
```
Comment (richsalz) [inline crypto/ui/ui_lib.c]: need explicit "!= NULL" test.

File: crypto/ui/ui_lib.c [hunk 2]
```diff
@@ -476,8 +483,16 @@ int UI_process(UI *ui)
...
-        return -1;
+    if (ui->meth->ui_close_session && !ui->meth->ui_close_session(ui)) {
+        if (state == NULL)
+            state = "closing session";
+        ok = -1;
+    }
+
+    if (ok == -1) {
+        UIerr(UI_F_UI_PROCESS, UI_R_PROCESSING_ERROR);
+        ERR_add_error_data(2, "while ", state);
```
Comment (richsalz) [inline crypto/ui/ui_lib.c]: ooh, clever.

File: crypto/ui/ui_lib.c [hunk 3]
```diff
@@ -427,9 +427,13 @@ static int print_error(const char *str, size_t len, UI *ui)
 int UI_process(UI *ui)
 {
     int i, ok = 0;
+    const char *state = NULL;
```
Comment (richsalz) [inline crypto/ui/ui_lib.c]: this should probably be something generic like "processing" to future-proof from future bunglers.

**Review by richsalz — APPROVED** [review]
two nits.  also okay to backport if you want.

File: crypto/ui/ui_lib.c (hunk 3 above)
Comment (levitte) [inline crypto/ui/ui_lib.c]: Hmmm, yeah ok

File: crypto/ui/ui_lib.c (hunk 1 above)
Comment (levitte) [inline crypto/ui/ui_lib.c]: Only there?  This is old stuff and that kind of change is needed all over the place.  I'd say that's for a separate cleanup PR.

**Comment by levitte** [comment]
Merged to master and 1.1.0

**Pull request closed without merging**